src/components/side_panel/chart/line_bar_pie_panel/config_panel.ts

```diff
@@ -137,6 +143,24 @@ export class LineBarPieConfigPanel extends Component<Props, SpreadsheetChildEnv>
     }
     return undefined;
   }
+
+  doesDataSetContainNegativeValues() {
+    let isNegative = false;
+    this.props.definition.dataSets.forEach((dataSet) => {
+      const getters = this.env.model.getters;
+      const sheetId = getters.getActiveSheetId();
+      const dataRange = createRange(getters, sheetId, dataSet)!;
+      const data = getters.getRangeValues(dataRange);
```

Comment:
this seems like a real hassle to recompute all of this all the time. and maybe a bad idea to expose the ranges (I realize it's already the case in `calculateHeaderPosition` but I'm questioning the relevance of such a move. in the case of `calculateHeaderPosition` I think we can manage the same thing with a  cal to `unboundedZone`.

 In this case, we are really just trying to access some information that is in the chart runtime. We might want to access directly instead of recomputing new ranges all the time.